src/NHibernate.Test/NHSpecificTest/GH1391/Fixture.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using NHibernate.Impl;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1391
+{
+	public class Fixture
+	{
+		[Test]
+		public void MultipleThreadsNested()
+		{
+			Parallel.For(0, 100, i =>
+			{
+				var guid = Guid.NewGuid();
+				using (new SessionIdLoggingContext(guid))
+				{
+					Assert.AreEqual(guid, SessionIdLoggingContext.SessionId);
```

Comment:
This is NUnit classic model of assertion. We are favoring the constraint model for new tests. See [two models](https://github.com/nunit/docs/wiki/Assertions#two-models).

Reply:
[Fixed](https://github.com/nhibernate/nhibernate-core/pull/1394/commits/b8198448c5f014e48962ab4b99787e5d28bb35a5#diff-1e66d0f683fb89f2119cd220860a4402R29)